dev/com.ibm.ws.com.unboundid/src/com/ibm/ws/com/unboundid/InMemoryLDAPServer.java

```diff
@@ -47,37 +58,79 @@
     private InMemoryDirectoryServerConfig config = null;
     private InMemoryDirectoryServer ds = null;
 
+    private static final String keystorePassword = "LDAPpassword";
+    private String keystore;
+
     /**
      * Creates a new instance of the in memory LDAP server. It initializes the directory
      * service.
      *
-     * @param bases The base entries to create for this in-memory LDAP server.
+     * @param useWimSchema      Asking the user if they want to use the default WIM schema
+     * @param bases             The base entries to create for this in-memory LDAP servers
+     * @param useSecureListener Use the LDAPS listener
      * @throws Exception If something went wrong
      */
-    public InMemoryLDAPServer(String... bases) throws Exception {
+    public InMemoryLDAPServer(boolean useWimSchema, boolean useSecureListener, String... bases) throws Exception {
 
         config = new InMemoryDirectoryServerConfig(bases);
         config.addAdditionalBindCredentials(getBindDN(), getBindPassword());
-        config.setListenerConfigs(
-                                  InMemoryListenerConfig.createLDAPConfig("LDAP", // Listener name
-                                                                          null, // Listen address. (null = listen on all interfaces)
-                                                                          0, // Listen port (0 = automatically choose an available port)
-                                                                          null) // StartTLS factory
-        ); // Client factory
 
+        keystore = extractResourceToFile("/resources/keystore.jks", "keystore", ".jks").getAbsolutePath();
+        final SSLUtil serverSSLUtil = new SSLUtil(new KeyStoreKeyManager(keystore, keystorePassword
+                        .toCharArray(), "JKS", "cert-alias"), new TrustAllTrustManager());
+        ArrayList<InMemoryListenerConfig> configs = new ArrayList<InMemoryListenerConfig>();
+        InMemoryListenerConfig secure = InMemoryListenerConfig.createLDAPSConfig("LDAPS", 0, serverSSLUtil.createSSLServerSocketFactory());
+        configs.add(secure);
+        InMemoryListenerConfig insecure = InMemoryListenerConfig.createLDAPConfig("LDAP", null, 0, null);
+        configs.add(insecure);
+        config.setListenerConfigs(configs);
+        Schema schema = null;
+        if (useWimSchema) {
+            InputStream in = getClass().getResourceAsStream("/resources/wimschema.ldif");
+            Schema wimschema = Schema.getSchema(in);
+            schema = Schema.mergeSchemas(Schema.getDefaultStandardSchema(), wimschema);
+        }
+        config.setSchema(schema);
+
+        ds = new InMemoryDirectoryServer(config);
+        if (useSecureListener)
```

Comment:
Can we not listen on both LDAP and LDAPS?

How would this effect getListenAddress()?